### DISCUSSION TITLE
with_expression in an aggregation

### DISCUSSION THREAD

**Discussion by ay-ramp** [sqlalchemy/sqlalchemy#10428] [Usage Questions]
Hey team, wonder if I could get some advice. I'm trying to do some aggregations on an ORM-mapped object that has some `query_expression` columns and not seeing a path forward. 

Here's a minimum reproducible example:

```
from typing import Any

from sqlalchemy import Column, Integer, create_engine, select, event, literal, func
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import (
    Session,
    query_expression,
    with_expression,
)

Base: Any = declarative_base()

class A(Base):
    __tablename__ = "a"
    id = Column(Integer, primary_key=True)
    x = query_expression("undefined")

e = create_engine("sqlite://", echo=False)
Base.metadata.create_all(e)

session = Session(e)

a = A(id=1)
session.add(a)

all_items_stmt = select(A).options(with_expression(A.x, literal(100)))
print(session.execute(all_items_stmt).scalars().one().x)


grouped_items = select(A.x, func.count(A.id)).group_by(A.x).options(
    with_expression(A.x, literal(100))
)
print(session.execute(grouped_items).all()) # <-- blows up
"""
  File "sqlalchemy/orm/strategy_options.py", line 1161, in _find_entity_prop_comparator
    raise sa_exc.ArgumentError(
sqlalchemy.exc.ArgumentError: Query has only expression-based entities, which do not apply to column property "A.x"

"""
```

(This example is obviously a little silly -- the thing I'm actually trying to do is to do some aggregations based on variable queries like `created_at - {customer_given_date}`.)

Using SQLAlchemy 1.4.49

Thanks in advance!

**Comment by zzzeek** [comment]
You can't use `with_expression()` here because this indicates that as you load new `A` objects, each `A` would have an attribute `x` that contains the literal value 100.   However, this query does not load `A` objects, it loads rows with individual non-object values in them, so there is nowhere to put "x".  this is a loader option only.

the query that would "work" here (although it doesn't make any sense to "group by" the value 100):

```py
expr = literal(100)
grouped_items = select(expr, func.count(A.id)).group_by(expr)
```

**Comment by ay-ramp** [comment]
Ah fair enough, I was overcomplicating it :) Thanks.
